collectors/node/node.go

```diff
@@ -162,9 +162,54 @@ func (c *Collector) buildDatapoints(in interface{}, t time.Time) []producers.Dat
 			continue
 		}
 
+		// Get the unit for the metric
+		var un = ""
+		uSlice := strings.Split(parsed.String(), ".")
+		uName := uSlice[len(uSlice)-1]
+		mName := uSlice[0]
+
+		switch mName {
+		case "memory":
+
+			switch uName {
+			case "free", "buffers", "cached":
+				un = "bytes"
+			default:
+				un = "count"
+			}
+
+		case "swap":
+			un = "bytes"
+
+		case "filesystem":
+			fsType := uName
+			if len(uSlice) >= 3 {
+				fsType = uSlice[len(uSlice)-2]
+			}
+
+			switch fsType {
+			case "capacity":
+				un = "bytes"
+			default:
+				un = "count"
+			}
+
+		case "network":
+
+			switch uName {
+			case "bytes":
+				un = "bytes"
+			default:
+				un = "count"
+			}
+
+		default:
+			un = "count"
+		}
```

Comment:
maybe instead of nested switch statements, we can make functions to extract the values base on input parameter? it'll be easier to unittest and read